openj9.test.sharedClasses.jvmti/src/test.sharedClasses.jvmti/net/openj9/stf/SharedClassesAPI.java

```diff
@@ -114,10 +115,14 @@ public void setUp(StfCoreExtension test, StfSharedClassesExtension sharedClasses
 		test.doMkdir("Create the cache directory", cacheDirLocation);
 		test.doMkdir("Create the config directory", configDirLocation);
 		
-		// To ensure we run from a clean state, attempt to destroy all persistent/non-persistent caches 
-		// from the default cache location which may have been left behind by a previous failed test.
-		sharedClasses.doDestroyAllPersistentCaches("Destroy Persistent Shared Classes Caches");
-		sharedClasses.doDestroyAllNonPersistentCaches("Destroy Non-Persistent Shared Classes Caches");
+		//We are running 5 Workloads namely WL1,..WL4. Each creates its own cache, so we should clean up each one of them
+		for (Tests apiTest : Tests.values()) {
+			for (int i = 1 ; i < 5 ; i++) {
+				String cacheName = apiTest.name() + "WL1" + i;
```

Comment:
Ideally the number of iterations and the cache name should be shared with the test and not hard coded separately. That way if the tests changes, the cleanup continues to be correct.

Reply:
Agree. I'll fix this in a separate PR.